src/pathfinder/npf/npf.cpp

```diff
@@ -137,7 +137,7 @@ static uint NPFDistanceTrack(TileIndex t0, TileIndex t1)
  *
  * @todo Think of a better hash.
  */
-static uint NPFHash(uint key1, uint key2)
+static uint NPFHash(TileIndex key1, uint key2)
 {
 	/* TODO: think of a better hash? */
 	uint part1 = TileX(key1) & NPF_HASH_HALFMASK;
```

Comment:
I can't comment on the line in question, but the cast on line 147 in `assert(IsValidTile((TileIndex)key1));` is no longer needed.